File: Github/tallerThreeC++/Transposed.cpp
#include <iostream>
#include <conio.h>
#include <stdlib.h>

using namespace std;
 
int main() 
{

	//Welcome message
	cout <<"Welcome\n";
	cout <<"this program print the matrix transpose m * m\n";
	cout <<"\nwrite the value of \"m\"\n";
	
	//all variables
	int m,i,j,k;
	cin >> m;	//read to size
	int board[m][m];
	int matrix [m][m];
	cout << "Enter the number "<<m*m<<" in the position:\n\n";
	
	//read the array
	for (i = 0; i < m; i++)
	{
		for (int j = 0; j < m; j++) 
		{
			cout << "board["<<i+1<<"]["<<j+1<<"]: ";
			cin >> board[i][j];
		}
	}
	
	cout<<"\nfirst matrix\n";
	//first matrix
	for (i = 0; i < m; i++)
	{
		cout<<"\n";
		for (j = 0; j < m; j++)
		{
			cout <<"["<<board[i][j]<<"]\t";
			matrix[j][i] = board[i][j];
		}
	}
	//print new matrix
	cout<<"\n\nnew matrix\n";
	for (i = 0; i < m; i++)
	{
		cout<<"\n";
		for (j = 0; j < m; j++)
		{
			cout <<"["<<matrix[i][j]<<"]\t";
		}
	}
	cout <<"\n\n";
	system("PAUSE");
	return 0;
}
//(rand() %101)-50

File: Github/TallerThreeJava/src/Transposed.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Transposed {
	
	//these are the objects to use
	static BufferedReader br = new BufferedReader (new InputStreamReader (System.in));
	static BufferedWriter bw = new BufferedWriter (new OutputStreamWriter (System.out));
	
	public static void main(String[]args) throws IOException
	{
		//all variables
		int m,i,j,k;
		String numbers;
		String[] array;
		int [][] board,transposed;
		
		//Welcome message
		bw.write( "Welcome\n"
				+ "this program print the matrix transpose m * m\n"
				+ "\nwrite the value of \"m\"\n");
		bw.flush();
		
		//read to size
		m = Integer.parseInt(br.readLine());
		
		//read the array 
		bw.write("\nEnter "+ m*m +" numbers separated by commas\n");
		bw.flush();
		
		numbers = br.readLine();
		array = numbers.split(",");
		board = new int [m][m];
		transposed = new int [m][m];
		
		k = 0;	//counter
		
		bw.write("\nfirst matrix\n");
		bw.flush();
		//first matrix
		for (i = 0; i < m; i++)
		{
			bw.write("\n");
			bw.flush();
			for (j = 0; j < m; j++)
			{
				board[i][j] = Integer.parseInt(array[k]);
				k++;
				bw.write("[" + board[i][j] + "]\t");
				bw.flush();
				
				//new matrix transposed
				transposed [j][i] = board[i][j];
			}
		}
		bw.write("\n\nnew matrix\n");
		bw.flush();
		
		//print new matrix
		for (i = 0; i < m; i++)
		{
			bw.write("\n");
			bw.flush();
			for (j = 0; j < m; j++)
			{
				bw.write("[" + transposed[i][j] + "]\t");
				bw.flush();
			}
		}
	}
}